### PR TITLE
cleanup: Modifies Wrapcheck linter

### PR DESCRIPTION
**Pull request by yati1998**
Wrapcheck is a  simple Go linter to check that errors
from external packages are wrapped during return to
help identify the error source during debugging.
This commit addresses the wrapcheck error

Updates:#2025

Signed-off-by: Yati Padia <ypadia@redhat.com>

### CODE REVIEW

**Comment by yati1998** [comment]
@Madhu-1 created seprate PR for easier merge and also have addressed the comments mentioned by you in https://github.com/ceph/ceph-csi/pull/2025

File: e2e/cephfs_helper.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func validateSubvolumegroup(f *framework.Framework, subvolgrp string) error {
 	cmd := fmt.Sprintf("ceph fs subvolumegroup getpath myfs %s", subvolgrp)
 	stdOut, stdErr, err := execCommandInToolBoxPod(f, cmd, rookNamespace)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to exec command in toolbox %w", err)
```
Comment (nixpanic) [inline e2e/cephfs_helper.go]: I think it is always good to split error reporting with a `:`, like:

```go
fmt.Errorf("failed to exec command in toolbox: %w", err)
```

This also makes `with error` unneeded in other new error returns.

File: e2e/configmap.go [hunk 2]
```diff
@@ -68,14 +68,14 @@ func createConfigMap(pluginPath string, c kubernetes.Interface, f *framework.Fra
...
-			return updateErr
+			return fmt.Errorf("failed to update configmap %w", updateErr)
 		}
 	}
 	if apierrs.IsNotFound(err) {
 		_, err = c.CoreV1().ConfigMaps(cephCSINamespace).Create(context.TODO(), &cm, metav1.CreateOptions{})
 	}
 
-	return err
+	return fmt.Errorf("error=%w", err)
```
Comment (nixpanic) [inline e2e/configmap.go]: This becomes problematic when `err == nil`

Comment (yati1998) [reply e2e/configmap.go]: True, but it shows wrap-check error. will check the alternative in this case.

File: e2e/configmap.go [hunk 3]
```diff
@@ -116,5 +116,5 @@ func createCustomConfigMap(c kubernetes.Interface, pluginPath string, subvolgrpI
 	cm.Namespace = cephCSINamespace
 	// since a configmap is already created, update the existing configmap
 	_, err = c.CoreV1().ConfigMaps(cephCSINamespace).Update(context.TODO(), &cm, metav1.UpdateOptions{})
-	return err
+	return fmt.Errorf("failed to update configmap: %w", err)
```
Comment (nixpanic) [inline e2e/configmap.go]: You will need to change this to

```go
if err != nil {
    return fmt.Errorf("failed to update configmap: %w", err)
}
return nil
```

This is what the e2e jobs cause to fail with:
```
Jun  3 09:32:57.401: failed to create configmap with error failed to update configmap: %!w(<nil>)
```

Comment (yati1998) [reply e2e/configmap.go]: okay got that.

Comment (humblec) [reply e2e/configmap.go]: @nixpanic when update fails, do we really get any error ?

Comment (yati1998) [reply e2e/configmap.go]: @humblec yes, after updating the code as mentioned above, we will get an error. Earlier even if the error was nil due to fmt.Errorf the create configmap was failing.

Comment (humblec) [reply e2e/configmap.go]: hmmm.. some object update errors are ignored from api server iic, any way does not hurt us to have the error return here.

**Review by nixpanic — DISMISSED** [review]

**Review by Madhu-1 — DISMISSED** [review]

**Review by Madhu-1 — DISMISSED** [review]

File: e2e/pod.go [hunk 4]
```diff
@@ -298,7 +298,7 @@ func deletePod(name, ns string, c kubernetes.Interface, t int) error {
 		}
 		e2elog.Logf("%s app  to be deleted (%d seconds elapsed)", name, int(time.Since(start).Seconds()))
 		if err != nil {
-			return false, err
+			return false, fmt.Errorf("failed to get pod: %w", err)
```
Comment (humblec) [inline e2e/pod.go]: @yati1998 we are interchangeably using `app` and `pod`, I would suggest to keep it in a consistent way like `app`

File: internal/journal/voljournal.go [hunk 5]
```diff
@@ -470,7 +470,7 @@ func reserveOMapName(ctx context.Context, monitors string, cr *util.Credentials,
 				continue
 			}
 
-			return "", err
+			return "", fmt.Errorf("failed to create object: %w", err)
```
Comment (humblec) [inline internal/journal/voljournal.go]: @yati1998  you can better place it as `failed to create omap object for `oMapNamePrefix+iterUUID`

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e/k8s-1.19

#1969 caused the issue:
```
Waiting up to 3m0s for all (but 0) nodes to be ready
```

**Pull request merged**